src/main/webapp/PermMarker.js

```diff
@@ -29,30 +35,39 @@ class PermMarker {
         position: coords,
         map: myMap.googleMap
       });
-    this.myInfoWindow = new MarkerInfoWindow(this);
+    this.editing = true;
     this.setMarkerListeners();
-    this.myInfoWindow.open();
+    myMap.openInfoWindow(this);
   }
 
   /**
    * Sets marker-triggered events
    */
   setMarkerListeners() {
-    this.googleMarker.addListener('dblclick', () => {
-      this.remove();
-      myMap.removeTempMarker();
-    });
     this.googleMarker.addListener('click', () => {
-      this.myInfoWindow.open();
-    });;
+      myMap.openInfoWindow(this);
+    });
   }
 
   /**
    * Removes the marker from the map
    */
   remove() {
-    myMap.deletePermMarker(this);
-    this.myInfoWindow.close();
     this.googleMarker.setMap(null);
   }
+
+  /** Returns whether this marker already has infowindow content*/
+  isEditing() {
+    return (!this.title && !this.body) || this.editing;
```

Comment:
It looks like having this.editing === false when both title and body are empty is an illegal state. Handling it this way can lead to bugs because other classes can set editing to false and assume that we're not in editing mode anymore.

Another way to fix this is to make editing private and have a setter that would throw an error if title and body are undefined. Can you do that instead and make this method just return this.editing? To make it private in JS, we just append an '_', so it becomes "this.editing_".

Reply:
Whoops the underscore were interpreted as italics in the last sentence. I was trying to say "append an underscore, so it becomes "this.editing_".

Reply:
Good point, I'll fix that asap.